### PR TITLE
add data structure exercises

### PR DESCRIPTION
**Pull request by juanbono**

### CODE REVIEW

File: calculate_bmi/solution/calculate_bmi.erl [hunk 1]
```diff
@@ -0,0 +1,20 @@
+-module(calculate_bmi).
+
+-export([bmi/1, classify/1]).
+
+-include("../src/person_record.hrl").
+
+bmi(P) when is_record(P, person) ->
+    P#person.weight / math:pow(P#person.height, 2).
+
+classify(P) ->
```
Comment (facundoolano) [inline calculate_bmi/solution/calculate_bmi.erl]: if you have a function that only does a case over the arguments, you most likely should use multiple function clauses instead

File: calculate_bmi/solution/calculate_bmi.erl [hunk 2]
```diff
@@ -0,0 +1,20 @@
+-module(calculate_bmi).
+
+-export([bmi/1, classify/1]).
+
+-include("../src/person_record.hrl").
```
Comment (facundoolano) [inline calculate_bmi/solution/calculate_bmi.erl]: prefer maps to recrods+header files. If you do end up using a record, it's better to encapsulate it in its own module, and always access it through functions. But seriously, maps.

File: merge_map/solution/merge_map.erl [hunk 3]
```diff
@@ -0,0 +1,14 @@
+-module(merge_map).
+
+-export([merge/2]).
+
+merge(Map1, Map2) ->
```
Comment (facundoolano) [inline merge_map/solution/merge_map.erl]: If maps:put is allowed, why not just:

```erlang
merge(Map1, Map2) ->
    maps:fold(fun(Key, V1, Map3) ->
                      maps:put(Key, V1, Map3)
              end, Map1, Map2).
``` 
Or, better:
```erlang
merge(Map1, Map2) -> maps:fold(fun maps:put/3, Map1, Map2).
```

Comment (juanbono) [reply merge_map/solution/merge_map.erl]: Thank you! I'll change that.

File: calculate_bmi/solution/calculate_bmi.erl (hunk 1 above)
Comment (juanbono) [reply calculate_bmi/solution/calculate_bmi.erl]: I'll change that!

File: calculate_bmi/solution/calculate_bmi.erl (hunk 2 above)
Comment (juanbono) [reply calculate_bmi/solution/calculate_bmi.erl]: [done :)](https://github.com/lambdaclass/erlings/pull/26/commits/61470a3dd16a1e5aa412d28c3d19cff3073d7754)

File: merge_map/solution/merge_map.erl (hunk 3 above)
Comment (juanbono) [reply merge_map/solution/merge_map.erl]: [done :)](https://github.com/lambdaclass/erlings/pull/26/commits/1c408cd55fbeaf47f8f6603a9e7f3c99039ba6b5)

**Pull request merged**